### PR TITLE
Fix #38 Edited Grunt File, Fixed Jshint Issues, updated submodules to latest versions

### LINKED ISSUE #38: Fix JSHint issues in bramble + bramble-extension code
Trying to run `grunt build` on our bramble branch fails with a million JSHint issues, mostly in our extension code.  We need to get this all cleaned up so we can pass lint on travis, and get grunt to work for the build target.

In code we own, we should fix things.  In code we don't own (e.g., slowparse) we should tell jshint to ignore the file completely.

### PR DESCRIPTION
**Pull request by KeniMorri**
i editied the grunt file so that jshint:src works fine, fixed general jshint issues around the program, and updated all submodules to latest versions with the jshint issues fixed.

### CODE REVIEW

File: src/filesystem/impls/makedrive/SampleFileSystem.js [hunk 1]
```diff
@@ -1,5 +1,4 @@
 /*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
-/*global define, appshell, $, window */
```
Comment (sedge) [inline src/filesystem/impls/makedrive/SampleFileSystem.js]: What's happening here? Why was this removed?

Comment (KeniMorri) [inline src/filesystem/impls/makedrive/SampleFileSystem.js]: From what I can tell, these are global declarations to JSHint to tell it that those Globals are going to be used within the confines of that file, but in SampleFileSystem.js none of these Globals are being used.

Comment (sedge) [inline src/filesystem/impls/makedrive/SampleFileSystem.js]: `define` is being used on line 5, so this should probably be added back in.

File: Gruntfile.js [hunk 2]
```diff
@@ -287,10 +287,20 @@ module.exports = function (grunt) {
             all: [
                 '<%= meta.grunt %>',
                 '<%= meta.src %>',
-                '<%= meta.test %>'
+                '<%= meta.test %>',
+                '!src/extensions/default/HTMLHinter/slowparse/**', // Not our code, so just ignore it
+                '!src/extensions/default/HTMLHinter/tooltipsy.source.js', // Klever has no idea how this works, so its ignored
```
Comment (sedge) [inline Gruntfile.js]: It looks like all of these (except makedrive-sync-icon) are references to third-party code of some kind. Your first comment should do it.

**Comment by sedge** [comment]
@KeniMorri This PR needs to be updated with a rebase from `bramble`.

**Comment by KeniMorri** [comment]
Ye definitely, i'm waiting on the pr for brackets live dev

File: Gruntfile.js [hunk 3]
```diff
@@ -287,10 +287,28 @@ module.exports = function (grunt) {
...
+                '<%= meta.src %>',
+                //This code is third party, thus we should not touch it
+                '!src/extensions/default/HTMLHinter/slowparse/**', 
+                //This javascript file has many third party methods, so skip it.
+                '!src/extensions/default/HTMLHinter/tooltipsy.source.js',
+                //Many Files are third party, thus we skip it
+                '!src/extensions/default/brackets-browser-livedev/nohost/**',
+                //This is a file we wrote, so we should lint it
+                'src/extensions/default/brackets-browser-livedev/nohost/src/NoHostServer.js'
+            ],
```
Comment (humphd) [inline Gruntfile.js]: So we have to explicitly name all these in order for them to get included in the main.js that require builds?

File: src/project/ProjectManager.js [hunk 4]
```diff
@@ -69,7 +69,6 @@ define(function (require, exports, module) {
         PerfUtils           = require("utils/PerfUtils"),
         FileUtils           = require("file/FileUtils"),
         FileSystemError     = require("filesystem/FileSystemError"),
-        Urls                = require("i18n!nls/urls"),
```
Comment (humphd) [inline src/project/ProjectManager.js]: This isn't code we own, how come this isn't an issue in upstream Brackets?

File: Gruntfile.js [hunk 5]
```diff
@@ -287,10 +287,28 @@ module.exports = function (grunt) {
             all: [
                 '<%= meta.grunt %>',
                 '<%= meta.src %>',
-                '<%= meta.test %>'
+                '<%= meta.test %>',
+                //This code is third party, thus we should not touch it
```
Comment (sedge) [inline Gruntfile.js]: One comment is enough:

`// These modules include lots of third-party code, so we skip them`

**Comment by sedge** [comment]
r+ with nit

**Pull request merged**